extensions/amp-accordion/1.0/test/test-accordion.js

```diff
@@ -119,18 +119,39 @@ describes.sandboxed('Accordion preact component', {}, (env) => {
       expect(sections.at(1).getDOMNode()).to.not.have.attribute('expanded');
       expect(sections.at(2).getDOMNode()).to.not.have.attribute('expanded');
 
+      const header0 = sections.at(0).find('header').getDOMNode();
+      const header1 = sections.at(1).find('header').getDOMNode();
+      const header2 = sections.at(2).find('header').getDOMNode();
+      const content0 = sections.at(0).find('div').getDOMNode();
+      const content1 = sections.at(1).find('div').getDOMNode();
+      const content2 = sections.at(2).find('div').getDOMNode();
+
       // Headers.
-      expect(sections.at(0).find('header').text()).to.equal('header1');
-      expect(sections.at(1).find('header').text()).to.equal('header2');
-      expect(sections.at(2).find('header').text()).to.equal('header3');
+      expect(header0.textContent).to.equal('header1');
+      expect(header1.textContent).to.equal('header2');
+      expect(header2.textContent).to.equal('header3');
 
       // Contents.
-      expect(sections.at(0).find('div').text()).to.equal('content1');
-      expect(sections.at(1).find('div').text()).to.equal('content2');
-      expect(sections.at(2).find('div').text()).to.equal('content3');
-      expect(sections.at(0).find('div').getDOMNode().hidden).to.be.false;
-      expect(sections.at(1).find('div').getDOMNode().hidden).to.be.true;
-      expect(sections.at(2).find('div').getDOMNode().hidden).to.be.true;
+      expect(content0.textContent).to.equal('content1');
+      expect(content1.textContent).to.equal('content2');
+      expect(content2.textContent).to.equal('content3');
+      expect(content0.hidden).to.be.false;
+      expect(content1.hidden).to.be.true;
+      expect(content2.hidden).to.be.true;
+
+      // Styling.
+      expect(header0.className.includes('section-child')).to.be.true;
```

Comment:
IMHO instead of testing for class names (or with them) I'd like to attach the accordion to DOM and call `getComputedStyle()` to check critical styles - the ones we marked with `!important`. Otherwise, the class  presence does not guarantee that we didn't make a bug with the class definition itself.

Reply:
Sounds good Dima.  For the `test-amp-accordion.js` (amp mode tests) I have made the updates.

For `test-accordion.js` (preact tests), I have not been able to get the tests to work.  A few things I tried include:

1. Attach the component DOM node to its own `ownerDocument.defaultView` (see lines 115 - 117).  However, I don't think the `style` tag from JSS is installed here, so not getting the styles as expected.

2. I also tried using options in the mount section (see line 109).  However, I'm unable to attach the component DOM node as `win.document` is apparently `undefined`.

I'm not too familiar with the scoping in the test environments.  Do you have any recommendations on where I can find out more about this?

Reply:
While these tests are super thorough and help increase confidence, it's not common to test CSS properties this granularly inside the components for visual styles. If it's not too much trouble, what do you think about copying the visual tests from 0.1 over? It should give us just as much information without as much of this tedium.

Reply:
Thanks!  As discussed offline, I've added an item in the tracker issue to look into the visual tests separately from this PR. https://github.com/ampproject/amphtml/issues/30445